### DISCUSSION TITLE
JSON responses being truncated at approx. 60Kb

### DISCUSSION THREAD

**Discussion by ollnixon** [vercel/community#3291] [Help] [answered]
### Summary

We have a strange issue where a deployment yesterday (17/07/2023) worked, but changes since do not work. Today's changes which were minor tweaks have been reverted (revert commits) so the code is identical, but we are still seeing issues.

Framework: Remix

Working deployment:
![CleanShot 2023-07-18 at 17 04 11](https://github.com/vercel/community/assets/118760/a6879fce-c654-4e41-aa0d-55438a413d6c)

Full response is 187,292 characters

Same code, new deployment:
![CleanShot 2023-07-18 at 17 05 37](https://github.com/vercel/community/assets/118760/45429ebc-a5ba-4057-8606-450925d24466)

Truncated response is 168,708 characters
![CleanShot 2023-07-18 at 17 08 57](https://github.com/vercel/community/assets/118760/550c9672-37df-41d8-bef8-21ae7f54b4d7)

Does anyone have any ideas? We're totally at a loss!

### Steps to Reproduce

n/a

**Comment by adamduren** [comment]
We are affected by this as well in production.

**Comment by danthareja** [comment]
@ollnixon - This is happening to me too!! I thought I was going crazy. I'm working on isolating a public repro case, but haven't been successful yet.

Have there been any recent changes to the inner workings of the `Content-Length` header (or similar)?

I'm running Remix 1.18 (no v2 feature flags on yet), with @vercel/remix - Zero code changes and seeing the same truncation start yesterday.

**If there's anything else I can do with my deployed instance to help debug, please let me know, this is a big showstopper for us.**

Here are some things I've noticed:

### Truncation happens with both JSON and text/html responses

#### json
![truncation](https://github.com/vercel/community/assets/6980359/9dc422a9-7aaf-4729-9d25-0db57e6ae61f)

#### html
![html](https://github.com/vercel/community/assets/6980359/f4ca18f1-a7da-4e0a-b54f-ae356d903ba0)


### It's not response size specific

![not size specific](https://github.com/vercel/community/assets/6980359/05a364c3-089a-43ec-9931-c9402644bbbc)

### It's not consistently failing for all Remix routes

In the above example,

`/new/peter-lougheed-provincial-park-elkwood` and `/new/jasper-national-park-whistlers` both route to the same `/new/$facility` route in Remix. The two requests share the same code, but only elkwood fails.

### When it does truncate, it always does in the same spot

In the above example,

`/new/peter-lougheed-provincial-park-elkwood` always stops at `https://reserve.a`

### I don't think it's a data issue

This is happening in both my prod and preview environments. Prod is backed by a Neon db, preview is backed by Supabase, both with the same contents.

**Comment by farchond** [comment]
We are also experiencing this issue - with 64kb being the truncation size limit. The issue is happening intermittently for us - it randomly works, though rarely.

We are logging the size of the json response as it's being returned from the server, as well as when it's received on the client - with a clear discrepancy.

We are using Remix version 1.16.

If there's anything we can do to help debug, let us know! Thanks.

**Comment by amyegan** [comment, marked as answer]
Thanks for reporting this, everyone! A fix has been released. Please redeploy your project to have the fix included in the new deployment, and let me know if you still have trouble with this 🙏 

Incident report: https://www.vercel-status.com/incidents/5c0cpdhr3n4p
